code/controllers/subsystem/job.dm

```diff
@@ -2,19 +2,25 @@ SUBSYSTEM_DEF(job)
 	name = "Jobs"
 	init_order = INIT_ORDER_JOBS
 	flags = SS_NO_FIRE
-
-	var/list/occupations = list()		//List of all potential jobs.
-	var/list/active_occupations	= list() //Jobs in use by the game mode.
+	var/ssjob_flags = NONE
+
+	var/list/occupations = list() //List of all jobs.
+	var/list/joinable_occupations = list() //List of jobs that can be joined as through the lobby.
+	var/list/joinable_occupations_by_category = list()
+	var/list/active_occupations	= list() //Jobs in use by the game mode at roundstart.
+	var/list/active_joinable_occupations = list() //Jobs currently joineable through the lobby (roundstart-only removed after).
+	var/list/active_joinable_occupations_by_category = list()
 	var/list/datum/job/name_occupations = list()	//Dict of all jobs, keys are titles.
 	var/list/type_occupations = list()	//Dict of all jobs, keys are types.
 
 	var/list/squads = list()			//List of potential squads.
 	var/list/active_squads = list()		//Squads being used by the game mode.
 
 	var/list/unassigned = list()		//Players who need jobs.
-	var/initial_players_to_assign = 0 	//Used for checking against population caps.
+	var/list/occupations_reroll //Jobs scaled up during job assignments.
+	var/initial_players_assigned = 0 	//Used for checking against population caps.
 
-	var/overflow_role = SQUAD_MARINE
+	var/datum/job/overflow_role = SQUAD_MARINE
```

Comment:
```suggestion
	var/datum/job/overflow_role = /datum/job/terragov/squad/standard
```

Reply:
SQUAD_MARINE is a string

Reply:
Turned it into a type that gets turned into the actual job instance's reference.